open-craft/openedx-completion-aggregator
# [MCKIN-7324] Provide API endpoint to fetch aggregators for all enrolled users in a course

**Description:** Alters the existing Completion Course Detail View to allow staff users to retrieve all aggregators for all enrolled users in a given course.

**JIRA:** MCKIN-7342, OC-4370

**Testing instructions:**

* In Studio, create 2 courses (e.g. Course A, Course B) with completable units.
* Enroll 2 learners in both courses (e.g. `honor`, `verified`)
* As the first learner, complete several units of Course A and Course B.
* As the second learner, complete units only in Course A.
* As a learner, check the Completion Aggregator API for each course to ensure that you can see only that learner's aggregations using the various URLs.
   Can add `?requested_fields=user` to see the username.
   * list all courses: `<lms_url>/completion-aggregator/v1/course/`
   * course detail: `<lms_url>/completion-aggregator/v1/course/<course_id>`
   * course detail with aggregators: `<lms_url>/completion-aggregator/v1/course/<course_id>/? requested_fields=sequential,chapter,vertical`
* As a `staff` user, check that you can see all the completions and aggregations for all the enrolled users using the detail URLs.
   * course detail for all users: `<lms_url>/completion-aggregator/v1/course/<course_id>`
   * course detail for single user: `<lms_url>/completion-aggregator/v1/course/<course_id>/?username=staff`
   * course detail with aggregators: `<lms_url>/completion-aggregator/v1/course/<course_id>/? requested_fields=sequential,chapter,vertical`
      Note that the requestable `user` field has been added automatically, to distinguish the different users.

**Reviewers:**

**Merge checklist:**
- [ ] <strike>Version bumped</strike> -- feature branch, will bump version when that merges to master.
- [ ] <strike>Changelog record added</strike> -- will do when version bumped
- [x] Documentation updated (not only docstrings) -- API documentation is generated from docstrings.

**Post merge:** - N/A, on feature branch
